### PR TITLE
feat: add trust_remote_code to hf_trainer_api examples

### PR DESCRIPTION
**Pull request by garrett361**
## Description

This small pr adds the ability to easily specify the `--trust_remote_code` option via the entrypoint in our `hf_trainer_ap` examples, e.g. as in [the line here](https://github.com/determined-ai/determined/blob/8d120e8ac2db019f0ab1c83a608a6f665f495a1b/examples/hf_trainer_api/hf_image_classification/adaptive.yaml#L49).  This flag is needed when running some models from huggingface, such as MosaicML's [mpt models](https://huggingface.co/mosaicml/mpt-7b) or the [falcon models](https://huggingface.co/tiiuae/falcon-40b-instruct) (though both require additional, more complex changes to run).

## Test Plan

Verify the existing `hf_trainer_ai` examples still run.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: examples/hf_trainer_api/hf_language_modeling/run_clm.py
Comment (aciborowska) [inline examples/hf_trainer_api/hf_language_modeling/run_clm.py:1]: I think we are missing `trust_remote_code` when instantiating the model in line 478 `model = AutoModelForCausalLM.from_pretrained(...)`

**Review by aciborowska — APPROVED** [review]
lgtm, with one question

File: examples/hf_trainer_api/hf_language_modeling/run_clm.py
Comment (garrett361) [reply examples/hf_trainer_api/hf_language_modeling/run_clm.py:1]: `trust_remote_code` gets passed into `config` in L445 which then gets passed into the above, so it should work ok. Used the same pattern in testing code elsewhere.

Comment (aciborowska) [reply examples/hf_trainer_api/hf_language_modeling/run_clm.py:1]: my bad, I missed `config`. but then do we need pass `trust_remote_code` in image_classification to `AutoModelForImageClassification` (line 333) if we are also passing config with `trust_remote_code`?

Comment (garrett361) [reply examples/hf_trainer_api/hf_language_modeling/run_clm.py:1]: ah yeah, there is redundancy there, will remove

**Pull request merged**